### DISCUSSION TITLE
Nextjs getStaticPaths + Firestore

### DISCUSSION THREAD

**Discussion by nonoumasy** [vercel/next.js#33935] [Help]
Just a question regarding Firestore and Nextjs. With getStaticPaths, nextjs needs to find all the IDs of each document and pass this as 'paths' to getStaticProps. My question is, how often does this happen? Since its accessing Firestore, this counts as 1 read per document and since this needs all the paths of a collection, it's basically all the documents in a collection. Does Nextjs keep a copy of this paths object somewhere to refer to until there is a new document added to the collection? Does Nextjs do this behind the scenes? I'm assuming its not doing this everytime you open a dynamic page?

Below is code I use to fetch all journal documents in Firestore.

```
export const getStaticPaths = async () => {
  const res = await getJournals()
  const paths = res.map((item) => {
    return {
      params: { id: item.id.toString() },
    }
  })

  return {
    paths,
    fallback: true,
  }
}

export const getStaticProps = async ({ params }) => {
  const journalData = await getJournal(params.id)
  const data1 = JSON.stringify(journalData)
  const entriesData = await getEntries(params.id)
  const data2 = entriesData.map((item) => JSON.stringify(item))

  return {
    props: {
      data1,
      data2,
    },
    revalidate: 1,
  }
}
```

**Comment by sethsnel** [comment]
Usually `getStaticPaths` is only ran during `npm run build/yarn build`. During `npm/yarn dev` getStaticpaths is also run.
It also depends a bit on what you configure for `fallback`.
https://nextjs.org/docs/api-reference/data-fetching/get-static-paths
